src/video/uikit/SDL_uikitviewcontroller.m

```diff
@@ -479,45 +470,53 @@ - (void)setKeyboardProperties:(SDL_PropertiesID) props
     } else {
         textField.enablesReturnKeyAutomatically = NO;
     }
-}
 
-// reveal onscreen virtual keyboard
-- (void)showKeyboard
-{
-    if (keyboardVisible) {
+    if (!textField.window) {
+        /* textField has not been added to the view yet,
+         we don't have to do anything. */
         return;
     }
 
-    keyboardVisible = YES;
-    if (textField.window) {
-        showingKeyboard = YES;
+    // the text field needs to be re-added to the view in order to update correctly.
+    UIView *superview = textField.superview;
+    [textField removeFromSuperview];
+    [superview addSubview:textField];
```

Comment:
This can technically be split to a separate PR but I figured I could include it in this PR since I'm touching the entire file anyway.